components/raftstore/src/store/peer_storage.rs

```diff
@@ -492,13 +524,8 @@ impl PeerStorage {
         );
         let raft_state = init_raft_state(&engines, region)?;
         let apply_state = init_apply_state(&engines, region)?;
-        if raft_state.get_last_index() < apply_state.get_applied_index() {
-            panic!(
-                "{} unexpected raft log index: last_index {} < applied_index {}",
-                tag,
-                raft_state.get_last_index(),
-                apply_state.get_applied_index()
-            );
+        if let Err(e) = validate_states(&raft_state, &apply_state) {
+            return Err(box_err!("{} validate state fail: {:?}", tag, e));
```

Comment:
I think raft_state::commit index should forward to apply_state::last_commit_index in here

Reply:
See L426.